ext/opentelemetry-ext-zipkin/src/opentelemetry/ext/zipkin/__init__.py

```diff
@@ -212,6 +212,17 @@ def _extract_tags_from_span(attr):
     return tags
 
 
+def _extract_tags_from_span(span: Span):
+    tags = _extract_tags_from_dict(span.attributes)
+    if span.resource:
+        resource_tags = _extract_tags_from_dict(span.resource.labels)
+        if tags:
+            tags.update(resource_tags)
+        else:
+            tags = resource_tags
```

Comment:
this is optimized over the case of just calling update, but how much does this really improve performance?

I would argue this is premature optimization, and we may want to opt for a more succinct codebase for now (just call update on tags, ensure it's always a dict).

Reply:
done